### PR TITLE
feat(id-card): updates to application

### PR DESCRIPTION
**Pull request by berglindoma13**
# ...

Attach a link to issue if relevant

## What

* Made changes to the way citizenship affects application, parents with foreign citizenship need to be able to apply for children but not for themselves
* Modified rules for age of application and screens that appear based on age
* Made changes to rules regarding II id cards and ID id cards
* Removed location choice
* finished second guardian functionality
* fixed alert messages based on which parent you are and if there are 2 parents that need to approve
* Moved ID cards to front of radio buttons
* Added alert when either ID card version is unavailable to the applicant
* made price messages dynamic based on applicant
* Modified Price choices from 12 down to 4
* Added dataprovider for parentB 

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Enhanced validation for parents based on children's citizenship status.
  - Improved form structure for ID card applications, making it more dynamic and context-aware.
  - Added new utility functions for better applicant data handling and presentation.
  - Updated action card field types for greater flexibility and specificity.
  - Introduced dynamic messaging capabilities based on applicant details.

- **Bug Fixes**
  - Corrected approval status handling for second guardian information.

- **Refactor**
  - Streamlined eligibility checks and applicant information retrieval processes.
  - Improved control flow for application review processes.

- **Chores**
  - Removed deprecated constants and types to enhance code clarity.

### CODE REVIEW

File: libs/application/templates/id-card/src/lib/IdCardTemplate.ts [hunk 1]
```diff
@@ -240,6 +292,10 @@ const IdCardTemplate: ApplicationTemplate<
             },
           ],
           actionCard: {
+            tag: {
+              label: 'YAYYY', // TODO
```
Comment (sigruntg) [inline libs/application/templates/id-card/src/lib/IdCardTemplate.ts:296]: Should maybe have a better text here, even though it's something that is TODO

File: libs/application/templates/id-card/src/lib/dataSchema.ts [hunk 2]
```diff
@@ -43,10 +43,11 @@ export const IdCardSchema = z.object({
     phoneNumber: z.string().refine((v) => isValidPhoneNumber(v) || v === '', {
       params: error.invalidValue,
     }),
+    approved: z.boolean().optional(),
   }),
   priceList: z.object({
     priceChoice: z.enum([Services.EXPRESS, Services.REGULAR]),
-    location: z.string(),
+    // location: z.string(),
```
Comment (sigruntg) [inline libs/application/templates/id-card/src/lib/dataSchema.ts:50]: Remove code that is commented out?

**Review by sigruntg — APPROVED** [review]
LGTM! Just few small comments <3

File: libs/application/templates/id-card/src/forms/Approved.ts [hunk 3]
```diff
@@ -1,77 +1,112 @@
...
       bottomButtonMessage: reviewConfirmation.general.bottomButtonMessage,
     }),
+    buildAlertMessageField({
+      id: 'uiForms.conclusionAlertInfo1',
+      title: '',
+      alertType: 'info',
+      message: reviewConfirmation.general.infoMessageText1,
+      marginBottom: 0,
+      condition: (formValue, externalData) => {
+        const applicantNationalId = getValueViaPath(
```
[the remark this answers is not in the document]
Comment (berglindoma13) [reply libs/application/templates/id-card/src/forms/Approved.ts:51]: reused some functions and simplified

File: libs/application/templates/id-card/src/lib/dataSchema.ts (hunk 2 above)
Comment (berglindoma13) [reply libs/application/templates/id-card/src/lib/dataSchema.ts]: fixed

File: libs/application/templates/id-card/src/lib/IdCardTemplate.ts (hunk 1 above)
Comment (berglindoma13) [reply libs/application/templates/id-card/src/lib/IdCardTemplate.ts]: removed

**Review by Toti91 — APPROVED** [review]
Application-system changes LGTM

**Review by disaerna — APPROVED** [review]
Code owned files look good 👍🏼

**Review by baering — APPROVED** [review]
Core file LGTM

**Pull request merged**